From b7f29c45aa3f58800e35ca040b7c321e5e67c899 Mon Sep 17 00:00:00 2001
From: Sebastien Boeuf <sebastien.boeuf@intel.com>
Date: Thu, 26 Feb 2015 15:31:39 +0100
Subject: [PATCH 20/20] Make 1280x960 as default video resolution

In sensor configuration file ov5648.h, we had 1296x736 before 1280x960
while the default resolution chosen, if not any matching resolution is
found, is the first one defined in the structure. I think we want a
default resolution having a ratio as close as possible to the full
frame ratio.
An example is when we ask for QCIF (from the HAL in continuous video
mode), resolution 352x288 can't match any of available resolutions,
that is 1280x960 and 1296x736, so it falls back to the default one.
In case the dafault one is 1280x960, I can see the ISP does not need
to crop as much as it is done if the default is 1296x736.

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
---
 drivers/media/i2c/atomisp/ov5648.h | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/media/i2c/atomisp/ov5648.h b/drivers/media/i2c/atomisp/ov5648.h
index 72dc69b..1c6fa2f 100644
--- a/drivers/media/i2c/atomisp/ov5648.h
+++ b/drivers/media/i2c/atomisp/ov5648.h
@@ -790,34 +790,34 @@ struct ov5648_resolution ov5648_res_still[] = {
 
 struct ov5648_resolution ov5648_res_video[] = {
 	{
-	 .desc = "ov5648_720P_30fps",
-	 .width = 1296,
-	 .height = 736,
-	 .fps = 30,
+	 .desc = "ov5648_1280x960_30fps",
+	 .width = 1280,
+	 .height = 960,
 	 .pix_clk_freq = 84,
+	 .fps = 30,
 	 .used = 0,
-	 .pixels_per_line = 2397,
-	 .lines_per_frame = 1186,
+	 .pixels_per_line = 2500,
+	 .lines_per_frame = 1126,
 	 .bin_factor_x = 2,
 	 .bin_factor_y = 2,
 	 .bin_mode = 1,
 	 .skip_frames = 3,
-	 .regs = ov5648_720p_30fps_2lanes,
+	 .regs = ov5648_1280x960_30fps_2lanes,
 	 },
 	{
-	 .desc = "ov5648_1280x960_30fps",
-	 .width = 1280,
-	 .height = 960,
-	 .pix_clk_freq = 84,
+	 .desc = "ov5648_720P_30fps",
+	 .width = 1296,
+	 .height = 736,
 	 .fps = 30,
+	 .pix_clk_freq = 84,
 	 .used = 0,
-	 .pixels_per_line = 2500,
-	 .lines_per_frame = 1126,
+	 .pixels_per_line = 2397,
+	 .lines_per_frame = 1186,
 	 .bin_factor_x = 2,
 	 .bin_factor_y = 2,
 	 .bin_mode = 1,
 	 .skip_frames = 3,
-	 .regs = ov5648_1280x960_30fps_2lanes,
+	 .regs = ov5648_720p_30fps_2lanes,
 	 },
 };
 
-- 
2.1.0

